extension/src/functions.rs

```diff
@@ -167,3 +173,223 @@ impl<T: Into<Vec<u8>>> Converter for DefaultConverter<T> {
+	};
+
+	enum AtLeastOneByte {}
+	impl Contains<Vec<u8>> for AtLeastOneByte {
+		fn contains(input: &Vec<u8>) -> bool {
+			input.len() > 0
+		}
+	}
+
+	enum LargerThan100 {}
+	impl Contains<u8> for LargerThan100 {
+		fn contains(input: &u8) -> bool {
+			*input > 100
+		}
+	}
+
+	enum MustBeEven {}
+	impl Contains<u8> for MustBeEven {
+		fn contains(input: &u8) -> bool {
+			*input % 2 == 0
+		}
+	}
+
+	#[test]
+	fn filterting_works() {
+		fn contains<C: Contains<T>, T>(input: T, expected: bool) {
+			assert!(C::contains(&input) == expected);
+		}
+		contains::<Everything, u32>(42, true);
+		contains::<Everything, Vec<u8>>(vec![1, 2, 3, 4], true);
+		contains::<Nothing, u32>(42, false);
+		contains::<Nothing, Vec<u8>>(vec![1, 2, 3, 4], false);
+		contains::<AtLeastOneByte, Vec<u8>>(vec![], false);
+		contains::<AtLeastOneByte, Vec<u8>>(vec![1], true);
+		contains::<AtLeastOneByte, Vec<u8>>(vec![1, 2, 3, 4], true);
+		contains::<LargerThan100, u8>(100, false);
+		contains::<LargerThan100, u8>(101, true);
+		contains::<MustBeEven, u8>(100, true);
+		contains::<MustBeEven, u8>(101, false);
+	}
+
+	#[test]
+	fn execute_dispatch_call_function_works() {
+		new_test_ext().execute_with(|| {
+			let call =
+				RuntimeCall::System(Call::remark_with_event { remark: "pop".as_bytes().to_vec() });
+			// Insert one byte due to the `RemoveFirstByte` conversion configuration of `DispatchCallEverythingFuncId`.
+			let encoded_call = [0u8.encode(), call.encode()].concat();
+			let mut default_env = MockEnvironment::default();
+			// Initialize mock environment with function execution.
+			let mut env = MockEnvironment::new(
+				DispatchCallEverthingFuncId::get(),
+				encoded_call.clone(),
+				MockExt::default(),
+			);
+			assert_eq!(env.charged(), default_env.charged());
+			assert!(matches!(Functions::execute(&mut env), Ok(Converging(0))));
+			// Check that the two environments charged the same weights.
+			assert_eq!(
+				env.charged(),
+				function_dispatch_call_weight(
+					&mut default_env,
+					encoded_call.len() as u32,
+					call,
+					env.ext().address().clone()
+				)
+			);
+		})
+	}
+
+	#[test]
+	fn execute_dispatch_call_function_invalid_input_works() {
+		new_test_ext().execute_with(|| {
+			// Invalid encoded runtime call.
+			let input = vec![0, 0, 99];
+			let mut env = MockEnvironment::new(
+				DispatchCallEverthingFuncId::get(),
+				input.clone(),
+				MockExt::default(),
+			);
+			let error = pallet_contracts::Error::<Test>::DecodingFailed.into();
+			let expected = <() as ErrorConverter>::convert(error, &mut env).err();
+			assert_eq!(Functions::execute(&mut env).err(), expected);
+			assert_eq!(env.charged(), read_from_buffer(input.len() as u32,));
+		})
+	}
+
+	#[test]
+	fn execute_read_state_function_works() {
+		let read = RuntimeRead::Ping;
+		// Insert one byte due to the `RemoveFirstByte` conversion configuration of `ReadStateEverythingFuncId`.
+		let encoded_read = [0u8.encode(), read.encode()].concat();
```

Comment:
Why not just create another function variant that doesnt require the additional byte to avoid all the noise in these tests?